test/e2e/singlecluster/appwrapper_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	awv1beta2 "github.com/project-codeflare/appwrapper/api/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	awtesting "sigs.k8s.io/kueue/pkg/util/testingjobs/appwrapper"
+	utiltestingjob "sigs.k8s.io/kueue/pkg/util/testingjobs/job"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("AppWrapper", func() {
+	const (
+		resourceFlavorName = "appwrapper-rf"
+		clusterQueueName   = "appwrapper-cq"
+		localQueueName     = "appwrapper-lq"
+	)
+
+	var (
+		ns *corev1.Namespace
+		rf *kueue.ResourceFlavor
+		cq *kueue.ClusterQueue
+		lq *kueue.LocalQueue
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "appwrapper-e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		rf = testing.MakeResourceFlavor(resourceFlavorName).
+			NodeLabel("instance-type", "on-demand").
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, rf)).To(gomega.Succeed())
+
+		cq = testing.MakeClusterQueue(clusterQueueName).
+			ResourceGroup(
+				*testing.MakeFlavorQuotas(resourceFlavorName).
+					Resource(corev1.ResourceCPU, "5").
+					Obj(),
+			).
+			Preemption(kueue.ClusterQueuePreemption{
+				WithinClusterQueue: kueue.PreemptionPolicyLowerPriority,
+			}).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+
+		lq = testing.MakeLocalQueue(localQueueName, ns.Name).ClusterQueue(cq.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, cq, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, rf, true)
+	})
+
+	ginkgo.It("Should admit workloads that fit", func() {
+		numPods := 2
+		aw := awtesting.MakeAppWrapper("appwrapper", ns.Name).
+			Component(utiltestingjob.MakeJob("job-0", ns.Name).
+				Request(corev1.ResourceCPU, "100m").
+				Parallelism(int32(numPods)).
+				Completions(int32(numPods)).
+				Suspend(false).
+				Image(util.E2eTestSleepImage, []string{"10s"}).
```

Comment:
I prefer to make it end immediately like after 10ms. Is the only issue that you want to assert that Appswrapper is running? I think from the integration PoV he most important thing is that it gets scheduled and completes successfully.

I would expect the creation of pods etc to be tested outside of Kueue.

Maybe we can add that asserts when proven to be important based on specific issues. 

Also, if we want to assert on the running Appswrapper and the pods we could have another test with longer timeout but after the asserts just kill the pods.

Reply:
Makes sense.  I'll revise as suggested later today.

Reply:
Test simplified as requested.  Runs a 10ms batch.Job.  Only checks for AppWrapper (a) being unsuspended and (b) completing successfully.

Reply:
sgtm